### PR TITLE
AC-221 - Bugfixes

### PR DESCRIPTION
**Pull request by tmarzeion**

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/PatientDashboardActivity.java [hunk 1]
```diff
@@ -72,36 +84,35 @@ public void onBackPressed() {
...
         return true;
     }
+
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         int id = item.getItemId();
         switch (id) {
             case R.id.actionDelete:
-                new PatientDAO().deletePatient(mPatient.getId());
+                new PatientDAO().deletePatient(Long.valueOf(mId));
```
Comment (AdamGrzybkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/PatientDashboardActivity.java]: use presenter to access Dao

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/PatientDashboardPagerAdapter.java [hunk 2]
```diff
@@ -41,31 +42,38 @@
...
+    PatientDashboardPagerAdapter(FragmentManager fm, String id) {
         super(fm);
-        mPatient = patient;
+        this.mPatientId = id;
     }
 
     @Override
     public Fragment getItem(int i) {
+
+        Patient patient = new PatientDAO().findPatientByID(mPatientId);
```
Comment (AdamGrzybkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/PatientDashboardPagerAdapter.java]: adapter is part of the view, so you should use presenter to access dao

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/PatientDashboardActivity.java (hunk 1 above)
Comment (tmarzeion) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/PatientDashboardActivity.java]: Fixed

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/PatientDashboardPagerAdapter.java (hunk 2 above)
Comment (tmarzeion) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/PatientDashboardPagerAdapter.java]: Fixed

**Pull request merged**